backends/sycl-ref/ceed-sycl-restriction.sycl.cpp

```diff
@@ -0,0 +1,477 @@
+      CeedInt       elem     = t_ind / elem_size;
+
+      value += u[loc_node + comp * elem_size * num_elem + elem * elem_size];
+    }
+    v[ind + comp * comp_stride] += value;
+  });
+  return CEED_ERROR_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+// Apply restriction
+//------------------------------------------------------------------------------
+static int CeedElemRestrictionApply_Sycl(CeedElemRestriction r, CeedTransposeMode t_mode, CeedVector u, CeedVector v, CeedRequest *request) {
+  Ceed ceed;
+  CeedCallBackend(CeedElemRestrictionGetCeed(r, &ceed));
+  CeedElemRestriction_Sycl *impl;
+  CeedCallBackend(CeedElemRestrictionGetData(r, &impl));
+  Ceed_Sycl *data;
+  CeedCallBackend(CeedGetData(ceed, &data));
+
+  // Get vectors
+  const CeedScalar *d_u;
+  CeedScalar       *d_v;
+  CeedCallBackend(CeedVectorGetArrayRead(u, CEED_MEM_DEVICE, &d_u));
+  if (t_mode == CEED_TRANSPOSE) {
+    // Sum into for transpose mode, e-vec to l-vec
+    CeedCallBackend(CeedVectorGetArray(v, CEED_MEM_DEVICE, &d_v));
+  } else {
+    // Overwrite for notranspose mode, l-vec to e-vec
+    CeedCallBackend(CeedVectorGetArrayWrite(v, CEED_MEM_DEVICE, &d_v));
+  }
+
+  // Restrict
+  if (t_mode == CEED_NOTRANSPOSE) {
+    // L-vector -> E-vector
+    if (impl->d_ind) {
+      // -- Offsets provided
+      CeedCallBackend(CeedElemRestrictionOffsetNoTranspose_Sycl(data->sycl_queue, impl, d_u, d_v));
+    } else {
+      // -- Strided restriction
+      CeedCallBackend(CeedElemRestrictionStridedNoTranspose_Sycl(data->sycl_queue, impl, d_u, d_v));
+    }
+  } else {
+    // E-vector -> L-vector
+    if (impl->d_ind) {
+      // -- Offsets provided
+      CeedCallBackend(CeedElemRestrictionOffsetTranspose_Sycl(data->sycl_queue, impl, d_u, d_v));
+    } else {
+      // -- Strided restriction
+      CeedCallBackend(CeedElemRestrictionStridedTranspose_Sycl(data->sycl_queue, impl, d_u, d_v));
+    }
+  }
+  // Wait for queues to be completed. NOTE: This may not be necessary
+  CeedCallSycl(ceed, data->sycl_queue.wait_and_throw());
+
+  if (request != CEED_REQUEST_IMMEDIATE && request != CEED_REQUEST_ORDERED) *request = NULL;
+
+  // Restore arrays
+  CeedCallBackend(CeedVectorRestoreArrayRead(u, &d_u));
+  CeedCallBackend(CeedVectorRestoreArray(v, &d_v));
+  return CEED_ERROR_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+// Blocked not supported
+//------------------------------------------------------------------------------
+int CeedElemRestrictionApplyBlock_Sycl(CeedElemRestriction r, CeedInt block, CeedTransposeMode t_mode, CeedVector u, CeedVector v,
+                                       CeedRequest *request) {
+  // LCOV_EXCL_START
+  Ceed ceed;
+  CeedCallBackend(CeedElemRestrictionGetCeed(r, &ceed));
+  return CeedError(ceed, CEED_ERROR_BACKEND, "Backend does not implement blocked restrictions");
+  // LCOV_EXCL_STOP
+}
```

Comment:
Should be removed - this check moved to the interface level

Reply:
When you say "this check moved to the interface level", do you mean that if `ApplyBlock` isn't set via `CeedSetBackendFunctionCpp`, it'll auto error-out saying that it can't do that?

Reply:
Exactly, with an appropriate message that CI interprets as SKIP